aptos-move/aptos-aggregator/src/delayed_field_extension.rs

```diff
@@ -2,347 +2,120 @@
+    pub fn try_add_delta(
         &mut self,
-        input: u128,
+        id: DelayedFieldID,
+        max_value: u128,
+        input: SignedU128,
         resolver: &dyn DelayedFieldResolver,
     ) -> PartialVMResult<bool> {
-        if input > self.max_value {
-            // We do not have to record the overflow.
-            // We record the delta that result in overflows/underflows so that when we compute the actual value
-            // of aggregator, we can figure out if the output of try_add/try_sub changes.
-            // When input exceeds max_value, we know that no matter what the starting value of the
-            // aggregator is, it always results in an overflow.
+        // No need to record or check or try, if input value exceeds the bound.
+        if input.abs() > max_value {
             return Ok(false);
         }
-        let math = BoundedMath::new(self.max_value);
-        self.read_last_committed_aggregator_value(resolver)?;
-        match &mut self.state {
-            AggregatorState::Create { value } => {
-                // If aggregator knows the value, add directly and keep the state.
-                match math.unsigned_add(*value, input) {
-                    Ok(new_value) => {
-                        *value = new_value;
-                        Ok(true)
-                    },
-                    Err(_) => Ok(false),
-                }
-            },
-            AggregatorState::Delta {
-                speculative_start_value,
-                delta,
-                history,
-            } => {
-                let cur_value = expect_ok(
-                    math.unsigned_add_delta(speculative_start_value.get_any_value()?, delta),
-                )?;
 
-                if math.unsigned_add(cur_value, input).is_err() {
-                    let overflow_delta =
-                        expect_ok(ok_overflow(math.unsigned_add_delta(input, delta)))?;
-
-                    // if value overflowed, we don't need to record it
-                    if let Some(overflow_delta) = overflow_delta {
-                        history.record_overflow(overflow_delta);
-                    }
-                    Ok(false)
-                } else {
-                    let new_delta =
-                        expect_ok(math.signed_add(delta, &SignedU128::Positive(input)))?;
-                    *delta = new_delta;
-                    history.record_success(new_delta);
-                    Ok(true)
+        match self.delayed_fields.entry(id) {
+            Entry::Vacant(entry) => {
+                let result = resolver.delayed_field_try_add_delta_outcome(
+                    &id,
+                    &SignedU128::Positive(0),
+                    &input,
+                    max_value,
+                )?;
+                if result {
+                    entry.insert(DelayedChange::Apply(DelayedApplyChange::AggregatorDelta {
+                        delta: DeltaWithMax::new(input, max_value),
```

Comment:
A new delayed change is being inserted here. Should this be Create or Apply here?

Reply:
Create is for when we are creating an aggregator.

here we are doing a first apply within this transaction , but it is required that aggregator already exists (i.e. there was a create in a transaction before it.